nuxeo-ai-similar-content/src/main/java/org/nuxeo/ai/similar/content/operation/FindSimilar.java

```diff
@@ -0,0 +1,165 @@
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.ai.cloud.CloudClient;
+import org.nuxeo.ai.enrichment.EnrichmentUtils;
+import org.nuxeo.ai.sdk.objects.TensorInstances;
+import org.nuxeo.ai.sdk.rest.client.API;
+import org.nuxeo.ai.sdk.rest.client.InsightClient;
+import org.nuxeo.ai.similar.content.services.SimilarContentService;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.automation.core.annotations.Context;
+import org.nuxeo.ecm.automation.core.annotations.Operation;
+import org.nuxeo.ecm.automation.core.annotations.OperationMethod;
+import org.nuxeo.ecm.automation.core.annotations.Param;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentRef;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.ecm.core.blob.ManagedBlob;
+import org.nuxeo.runtime.api.Framework;
+
+@Operation(id = FindSimilar.ID, category = "Insight", label = "Find duplicate documents")
+public class FindSimilar {
+
+    private static final Logger log = LogManager.getLogger(FindSimilar.class);
+
+    public static final String ID = "Insight.FindSimilar";
+
+    @Context
+    protected CoreSession session;
+
+    @Context
+    protected CloudClient client;
+
+    @Context
+    protected SimilarContentService scs;
+
+    @Param(name = "xpath", required = false)
+    protected String xpath = FILE_CONTENT;
+
+    @Param(name = "max", required = false)
+    protected long max;
+
+    @OperationMethod
+    public List<DocumentModel> run(DocumentModel doc) throws OperationException {
+        if (!scs.anyMatch(doc)) {
+            log.debug("None dedup filters fit document {}", doc.getId());
+            return emptyList();
+        }
+
+        List<String> ids = findSimilarIds(doc, null);
+        return resolveDocuments(ids);
+    }
+
+    @OperationMethod
+    public List<DocumentModel> run(Blob blob) throws OperationException {
+        if (blob.getLength() >= Long.parseLong(
+                Framework.getProperty(AI_BLOB_MAX_SIZE_CONF_VAR, AI_BLOB_MAX_SIZE_VALUE))) {
+            throw new OperationException("Blob is to large; size = " + blob.getLength());
+        }
+
+        List<String> ids = findSimilarIds(null, blob);
+        return resolveDocuments(ids);
+    }
+
+    protected List<String> findSimilarIds(@Nullable DocumentModel doc, @Nullable Blob blob) throws OperationException {
+        Optional<InsightClient> client = this.client.getClient(session);
+        if (client.isEmpty()) {
+            throw new OperationException(
+                    "Could not obtain Insight Client for user " + session.getPrincipal().getActingUser());
+        }
+
+        String id = doc == null ? null : doc.getId();
+        TensorInstances tensor = constructTensor((ManagedBlob) blob, id);
```

Comment:
Something to discuss: Here we can have a null blob  (so a null tensor instance) but at the same time, we are having the xpath so we could get a potential blob from there. On the other operations im doing, i don't take any blob as input, always the doc/xpath. Something maybe to sync together.